projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/ingest_to_impala.py

```diff
@@ -0,0 +1,87 @@
+from vdk.internal.builtin_plugins.run.job_context import JobContext
+from vdk.internal.core import errors
+
+_log = logging.getLogger(__name__)
+
+
+class IngestToImpala(IIngesterPlugin):
+    def __init__(self, context: JobContext):
+        self._context = context
+
+    def ingest_payload(
+        self,
+        payload: List[dict],
+        destination_table: Optional[str],
+        target: Optional[str] = None,
+        collection_id: Optional[str] = None,
+    ) -> None:
+        """
+        See parent class doc for details
+        """
+
+        _log.info(
+            f"Ingesting payloads to table: {destination_table} in Impala database; "
+            f"collection_id: {collection_id}"
+        )
+
+        with self._context.connections.open_connection(
+            "IMPALA"
+        ).connect() as connection:
+            cursor = connection.cursor()
+            query, parameters = self._populate_query_parameters_tuple(
+                destination_table, cursor, payload
+            )
+
+            try:
+                cursor.execute(query, parameters)
+                connection.commit()
+                _log.debug("Payload was ingested.")
+            except Exception as e:
+                errors.log_and_rethrow(
+                    errors.find_whom_to_blame_from_exception(e),
+                    _log,
+                    "Failed to send payload",
+                    "Unknown error. Error message was : " + str(e),
+                    "Will not be able to send the payload for ingestion",
+                    "See error message for help ",
+                    e,
+                    wrap_in_vdk_error=True,
+                )
+
+    @staticmethod
+    def _populate_query_parameters_tuple(
+        destination_table: str, cursor: PEP249Cursor, payload: List[dict]
+    ) -> (str, Tuple[str]):
+        """
+        Returns insert into destination table tuple of query and parameters;
+        E.g. for a table dest_table with columns val1, val2 and payload size 3, this method will return:
+        'INSERT INTO dest_table (val1, val2) VALUES (%s, %s), (%s, %s), (%s, %s)', ['val1', 'val2']
+
+        :param destination_table: str
+            the name of the destination table
+        :param cursor: PEP249Cursor
+            the database cursor
+        :param payload: List[dict]
+            the payloads to be ingested
+        :return: Tuple[str, Tuple[str]]
+            tuple containing the query and parameters
+        """
+        cursor.execute(f"SELECT * FROM {destination_table} WHERE false")
+        columns = [c[0] for c in cursor.description]
+
+        row_placeholder = f"({', '.join('%s' for column in columns)})"
+
+        return (
+            f"INSERT INTO {destination_table} ({', '.join(columns)}) "
+            f"VALUES {', '.join([row_placeholder for i in range(len(payload))])}",
+            tuple(obj[column] for obj in payload for column in columns),
```

Comment:
What would happen if a column is missing from an obj of the payload? Should we safeguard against such situations, or do we assume it is not possible for the payload to be malformed?

Reply:
I don't believe we check for this in any of the ingest module in DB plugins - as mentioned in the PR description, this code is almost 1 to 1 with the ingest module in vdk-greenplum.

Reply:
I see. Ok.